src/Microsoft.OData.Core/EdmExtensionMethods.cs

```diff
@@ -152,17 +154,150 @@ public static bool TryParse(this IEdmEnumType enumType, long value, out IEdmEnum
+        /// <returns>True if the member name exists in the enum type; otherwise, false.</returns>
+        public static bool HasMember(this IEdmEnumType enumType, ReadOnlySpan<char> memberName, StringComparison comparison = StringComparison.Ordinal)
         {
             foreach (IEdmEnumMember member in enumType.Members)
             {
-                if (string.Equals(member.Name, memberName, comparison))
+                if (memberName.Equals(member.Name.AsSpan(), comparison))
                 {
                     return true;
                 }
             }
 
             return false;
         }
+
+        /// <summary>
+        /// Checks if the given member name exists in the enum type.
+        /// </summary>
+        /// <param name="enumType">The enum type to search for the member.</param>
+        /// <param name="memberName">The name of the member to locate, represented as a <see cref="ReadOnlySpan{T}"/> of characters.</param>
+        /// <param name="exactMemberName">When this method returns, contains the <see cref="IEdmEnumMember"/> that matches the specified name, if a match is found; otherwise, null.</param>
+        /// <param name="comparison">The <see cref="StringComparison"/> to use when comparing the member names. The default is <see cref="StringComparison.Ordinal"/>.</param>
+        /// <returns>True if the member name exists in the enum type; otherwise, false.</returns>
+        public static IEdmEnumMember FindMember(this IEdmEnumType enumType, ReadOnlySpan<char> memberName, StringComparison comparison = StringComparison.Ordinal)
+        {
+            foreach (IEdmEnumMember member in enumType.Members)
+            {
+                if (memberName.Equals(member.Name.AsSpan(), comparison))
+                {
+                    return member;
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Parses the specified integral value into a comma-separated string of flag names based on the members of the given EDM enum type.
+        /// </summary>
+        /// <param name="enumType">The EDM enum type containing the members to parse.</param>
+        /// <param name="value">The integral value to parse into flag names.</param>
+        /// <returns>A comma-separated string of flag names corresponding to the set bits in the specified value. Returns null otherwise.</returns>
+        public static string ParseFlagsFromIntegralValue(this IEdmEnumType enumType, long value)
+        {
+            var result = new List<string>();
+            long remaining = value;
+
+            for (int index = enumType.Members.Count() - 1; index >= 0; index--)
+            {
+                IEdmEnumMember member = enumType.Members.ElementAt(index);
+                long flagValue = Convert.ToInt64(member.Value.Value);
+                if (flagValue != 0 && (remaining & flagValue) == flagValue)

```

Comment:
If the enum defines a zero-valued member (often "None"), will it not be excluded by this condition? Think about how that should be handled